floodwatch/src/js/components/Generate.js

```diff
@@ -1,54 +1,95 @@
-import d3 from 'd3';
-// import _ from 'lodash';
+//@flow
+
+import type {Filter, FilterLogic} from './filtertypes'
+import type {PersonResponse, FilterRequestItem} from '../api/types';
+import type {VisibilityMap} from './Compare'
+
+import {getVisibilityMap, generateDifferenceSentence, createSentence} from './comparisontools';
+import DemographicKeys from '../../stubbed_data/demographic_keys.json';
 import React, { Component } from 'react';
 import { Row, Col } from 'react-bootstrap';
 import { Chart } from './Chart';
 import {render} from 'react-dom';
 import url from 'url';
+import _ from 'lodash';
 
-import {getVisibilityMap, generateDifferenceSentence, createSentence} from './comparisontools';
-import type {VisibilityMap} from './Compare'
 
+type StateType = {
+  filterA: any,
+  filterB: any,
+  dataA: any,
+  dataB: any,
+  curTopic: string,
+  visibilityMap: VisibilityMap,
+  lSentence: string,
+  rSentence: string,
+  sentence: string
+}
 
-export class Generate extends Component {
-  render() {
-    let curData;
-    if (url.parse(window.location.href, true).query.data) {
-        curData = JSON.parse(url.parse(window.location.href, true).query.data);
-    }
+function initialState(): StateType {
+  let curData = null;
+  if (url.parse(window.location.href, true).query.data) {
+    curData = JSON.parse(url.parse(window.location.href, true).query.data);
+  } else {
+    curData = {}
+  }
+
+  let visibilityMap = getVisibilityMap(curData.dataA, curData.dataB)
+  const lVal = curData.curTopic ? curData.dataA[curData.curTopic] : 0;
+  const rVal = curData.curTopic ? curData.dataB[curData.curTopic] : 0;
+
+  const lSentence = createSentence(decodeFilterRequestItem(curData.filterA));
+  const rSentence = createSentence(decodeFilterRequestItem(curData.filterB));
 
-    let visibilityMap = getVisibilityMap(curData.dataA, curData.dataB)
-    const lVal = curData.curTopic ? curData.dataA[curData.curTopic] : 0;
-    const rVal = curData.curTopic ? curData.dataB[curData.curTopic] : 0;
+  const sentence = generateDifferenceSentence(curData.filterA, curData.filterB, lVal, rVal, curData.curTopic)
 
-    const lSentence = createSentence(curData.filterA);
-    const rSentence = createSentence(curData.filterB);
+  return {
+    dataA: (curData.dataA) ? curData.dataA : {},
+    dataB: (curData.dataB) ? curData.dataB : {},
+    filterA: (curData.filterA) ? curData.filterA : {},
+    filterB: (curData.filterB) ? curData.filterB : {},
+    curTopic: (curData.curTopic) ? curData.curTopic : null,
+    visibilityMap: visibilityMap,
```

Comment:
ES2016 feature: you don't have to repeat same-named keys and variables, so this and the next few can just be:

```js
{
  ...
  lSentence,
  rSentence,
  sentence
}
```